client/service/src/task_manager/mod.rs

```diff
@@ -311,7 +313,13 @@ impl TaskManager {
 		Box::pin(async move {
 			join_all(children_shutdowns).await;
 			completion_future.await;
-			drop(keep_alive);
+
+			// The keep_alive stuff is holding references to some RPC handles etc. These
+			// RPC handles spawn their own tokio stuff and that doesn't like to be closed in an
+			// async context. So, we move the deletion to some other thread.
```

Comment:
how did it work previously? or is this a change in behavior from tokio 0.2 to 1.0?

Reply:
I think tokio 1.0 can only detect tokio 1.0. So it did not detected the shutdown being async before. And as both used 0.1 or whatever, tokio itself provided an async function.

This is entirely some hackery stuff in here. When we refactor all of this, there should not be any "keep_alive", because this itself is just a hack...